File: zero judge/a020 身分證檢驗.cpp
#include<bits/stdc++.h>
using namespace std;
int main() {
	string a,s="==========ABCDEFGHJKLMNPQRSTUVXYWZIO";
	cin>>a;
	int x,ans=a[(a.size()-1)]-'0';
	for(int i=a.size()-2,j=1;i>0;i--,j++){
		ans+=(a[i]-'0')*j;
	}
	for(int i=10;i<s.size();i++){
		if(a[0]==s[i]){
			x=i;
			break;
		}
	}
	ans+=((x/10)+(x%10*9));
	if(ans%10==0){
		cout<<"real\n";
	}else{
		cout<<"fake\n";
	}
	return 0;
}

File: zero judge/a020. 身分證檢驗.py
d = {'A':1,'B':10,'C':19,'D':28,'E':37,'F':46,'G':55,'H':64,'I':39,'J':73,'K':82,'L':2,'M':11,'N':20,'O':48,'P':29,'Q':38,'R':47,'S':56,'T':65,'U':74,'V':83,'W':21,'X':3,'Y':12,'Z':30}
s = input()
ans = d[s[0]]
for i in range(1,9):
	ans += int(s[i]) * (9 - i)
ans = (ans + int(s[-1])) % 10
print('fake' if ans else 'real')